src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp

```diff
@@ -361,6 +361,44 @@ class MarkDivWithEps : public MatcherPass {
     }
 };
 
+// todo: description
+class GatherHandling : public MatcherPass {
+public:
+    OPENVINO_RTTI("GatherHandling", "0");
```

Comment:
Do we need this transformation anymore? Looks like to solve the problem with extra RAM CVS-127678 it should be enough just to remove Gather/GatherBase from the list above.

Reply:
Pow is type sensitive op. It means that we propagate fp32 to pow inputs, also gather became fp32. Transformation mark gather and it's inputs as eligible for fp16 compression. Check the test

Reply:
I've commented GatherHandling transformation 
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/06e1e94c-e7a0-4cbf-b975-ae9005314b0f)

and got the expected subgraph with fp16 Gather operation
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/451cc4c5-61bb-4555-950e-40d71de76976)

the test passes if I comment these 2 lines 
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/371a89ec-4d4f-4c69-84ac-c270c7ed1a3e)

Reply:
Have removed